client/test/commands/createSmartContractProjectCommand.test.ts

```diff
@@ -0,0 +1,265 @@
+        await vscode.commands.executeCommand('blockchain.createSmartContractProjectEntry');
+        spawnStub.should.have.been.calledWith('/bin/sh', ['-c', 'yo fabric:chaincode < /dev/null']);
+        errorSpy.should.have.been.calledWith('Issue creating smart contract project');
+    });
+
+    it('should should not do anything if the user cancels the open dialog', async () => {
+        // We actually want to execute the command!
+        sendCommandStub.restore();
+
+        const originalSpawn = child_process.spawn;
+        const spawnStub: sinon.SinonStub = mySandBox.stub(child_process, 'spawn');
+        spawnStub.withArgs('/bin/sh', ['-c', 'yo fabric:chaincode < /dev/null']).callsFake(() => {
+            return originalSpawn('/bin/sh', [ '-c', 'echo blah && echo "  Go" && echo "  JavaScript" && echo "  TypeScript  [45R"']);
+        });
+        quickPickStub.onCall(0).returns('TypeScript');
+
+        openDialogStub.resolves(undefined);
+        await vscode.commands.executeCommand('blockchain.createSmartContractProjectEntry');
+        openDialogStub.should.have.been.calledOnce;
+        spawnStub.should.have.been.calledWith('/bin/sh', ['-c', 'yo fabric:chaincode < /dev/null']);
+        executeCommandStub.should.have.been.calledOnce;
+        executeCommandStub.should.have.not.been.calledWith('vscode.openFolder');
+        errorSpy.should.not.have.been.called;
+    });
+
+    it('should create a go chaincode project when the user selects go as the language', async () => {
+        sendCommandStub.restore();
+        try {
+            await mkdir(uri.fsPath);
+        } catch (error) {
+            if (!error.message.includes('file already exists') ) {
+                throw new error('failed to create test directory:' + uri.fsPath);
+            }
+        }
+        const originalSpawn = child_process.spawn;
+        const spawnStub: sinon.SinonStub = mySandBox.stub(child_process, 'spawn');
+        spawnStub.withArgs('/bin/sh', ['-c', 'yo fabric:chaincode < /dev/null']).callsFake(() => {
+            return originalSpawn('/bin/sh', [ '-c', 'echo blah && echo "  Go" && echo "  JavaScript" && echo "  TypeScript  [45R"']);
+        });
+        quickPickStub.onCall(0).returns('Go');
+        openDialogStub.resolves(uriArr);
+
+        await vscode.commands.executeCommand('blockchain.createSmartContractProjectEntry');
+        const pathToCheck = path.join(uri.fsPath, 'main.go');
+        const chaincodeExists = await fs_extra.pathExists(pathToCheck);
+        chaincodeExists.should.be.true;
+        spawnStub.should.have.been.calledWith('/bin/sh', ['-c', 'yo fabric:chaincode < /dev/null']);
+        executeCommandStub.should.have.been.calledTwice;
+        executeCommandStub.should.have.been.calledWith('vscode.openFolder', uriArr[0], true);
+        errorSpy.should.not.have.been.called;
+    }).timeout(20000);
+
+    it('should create a default javascript project if getting the available chaincode lanaguages fails', async () => {
```

Comment:
It doesn't seem safe to assume that javascript is available if you can't get the list of available projects (see above)